stellar/contract/round/src/internal.rs

```diff
@@ -0,0 +1,802 @@
+        );
+        assert!(
+            round_detail.end_time > round_detail.application_start_time,
+            "Round end time must be greater than round application start time"
+        );
+        assert!(round_detail.amount > 0, "Amount must be greater than 0");
+        assert!(
+            !round_detail.admins.is_empty(),
+            "Round admins must not empty"
+        );
+        assert!(
+            round_detail.admins.len() < 5,
+            "Round admins must be less than 5"
+        );
+
+        let round_init = is_initialized(env);
+
+        assert!(!round_init, "Round already initialized");
+
+        let mut num_picks_per_voter = 2;
+
+        if round_detail.num_picks_per_voter.is_some() {
+            num_picks_per_voter = round_detail.num_picks_per_voter.unwrap();
+            assert!(
+                num_picks_per_voter > 0,
+                "Number of picks per voter must be greater than 0"
+            );
+            assert!(
+                num_picks_per_voter <= 10,
+                "Number of picks per voter must be less than or equal 10"
+            );
+        }
+
+        assert!(
+            round_detail.contact.len() <= 10,
+            "Contact must be less than 10"
+        );
+        assert!(
+            round_detail.image_url.len() <= 200,
+            "Image URL must be less than 200 characters. Use IPFS Hash Only"
+        );
+
+        let round_info = RoundDetail {
+            id: round_detail.id,
+            name: round_detail.name,
+            description: round_detail.description,
+            start_time: round_detail.start_time,
+            end_time: round_detail.end_time,
+            image_url: round_detail.image_url,
+            contact: round_detail.contact,
+            owner,
+            admins: round_detail.admins,
+            application_start_time: round_detail.application_start_time,
+            application_end_time: round_detail.application_end_time,
+            expected_amount: round_detail.amount,
+            is_completed: false,
+            use_whitelist: round_detail.use_whitelist.unwrap_or(false),
+            num_picks_per_voter,
+        };
+
+        write_round_info(env, &round_info);
+        write_token_address(env, &token_address);
+        write_project_contract(env, &registry_address);
+        log_create_round(env, round_info);
+    }
+
+    fn change_voting_period(env: &Env, admin: Address, round_start_time: u64, round_end_time: u64) {
+        admin.require_auth();
+
+        assert!(
+            round_start_time < round_end_time,
+            "Round start time must be less than round end time"
+        );
+
+        let mut round = read_round_info(env);
+
+        if round.owner != admin {
+            round
+                .admins
+                .iter()
+                .find(|round_admin| round_admin == &admin)
+                .expect("Only round owner or round admin can change voting period");
+        }
+
```

Comment:
style: can simplify this with single assert!

Reply:
this check is duplicated across many functions, can we create a single function to check this?